### PR TITLE
Forking the DataDog .NET Tracer to seed the OTel .NET In-Proc-Auto-Instrumentation-Agent.

### PR DESCRIPTION
**Pull request by macrogreg**
This is the initial seed for the official Open Telemetry .NET Tracer (aka in-process agent, aka auto-instrumentation agent).
We are doing this by merging the current master of _https://github.com/DataDog/dd-trace-dotnet_ into an empty master of _https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation_.

### CODE REVIEW

**Comment by macrogreg** [comment]
**Please do not merge before the Product folks conformed that all legal points have been addressed!**

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+* @DataDog/apm-dotnet
```
Comment (cijothomas) [inline .github/CODEOWNERS:1]: are you planning to merge this as is? or do more cleanups before merge?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- Instrumentation mode: [e.g. automatic with msi installer or manual with NuGet package]
+- Tracer version: [e.g. 1.0.0]
+- OS: [e.g. Windows Server 2012 R2 ]
+- CLR: [e.g. .NET Framework 4.6.2, .NET Core 2.1]
+
+**Additional context**
+Add any other context about the problem here.
+
+<!--
+NOTE: To contact our support team, see https://www.datadoghq.com/support/
```
Comment (cijothomas) [inline .github/ISSUE_TEMPLATE/bug_report.md:33]: Will you be removing all vendor specific contents?

File: .github/CODEOWNERS (hunk 1 above)
Comment (macrogreg) [reply .github/CODEOWNERS:1]: As discussed in the SIG meeting, I want to merge as is to unblock others from contributing. However, I do plan to clean up subsequently.

**Comment by macrogreg** [comment]
I just did a clean-up of the DD code. I will close this and re-create a PR with cleaned up code shortly.

**Pull request closed without merging**